src/upgrade/upgradeManager.ts

```diff
@@ -63,26 +63,33 @@ class UpgradeManager {
     }
 
     private static async checkUpgradableComponents(folder: WorkspaceFolder) {
-        if (!await languageServerApiManager.ready()) {
-            return;
-        }
-        const hasJavaError: boolean = await Jdtls.checkImportStatus();
-        if (hasJavaError) {
-            return;
-        }
+        return (instrumentOperation("upgradeManager.checkUpgradableComponents",
+            async (operationId: string) => {
+                if (!await languageServerApiManager.ready()) {
+                    return;
+                }
+                const hasJavaError: boolean = await Jdtls.checkImportStatus();
+                if (hasJavaError) {
+                    return;
+                }
 
-        const projectIssues: UpgradeIssue[] = [];
-        const uri = folder.uri.toString();
-        const projects = await Jdtls.getProjects(uri);
-        await Promise.allSettled(projects.map(async (projectNode) => {
-            const issues = await assessmentManager.getProjectIssues(projectNode);
-            projectIssues.push(...issues);
-        }));
+                const projectIssues: UpgradeIssue[] = [];
+                const uri = folder.uri.toString();
+                const projects = await Jdtls.getProjects(uri);
+                await Promise.allSettled(projects.map(async (projectNode) => {
+                    const issues = await assessmentManager.getProjectIssues(projectNode);
+                    projectIssues.push(...issues);
+                    sendInfo(operationId, {
```

Comment:
Seems will send multi info events in single checkUpgradableComponents? Shall we move it to L89?